### PR TITLE
Use ElementTree for XML parsing

### PR DESCRIPTION
**Pull request by nre-ableton**
The xmltodict package was not really great. Accessing the children of
an element was a bit dangerous since we could not be sure whether they
would be packaged in a list or not. Also, this library is a third-
party dependency and thus required pipenv to actually run the script.

That said, we still need pipenv because of third-party package
requirements for the fetch_jars.py script.

---

This was much easier than I thought... I feel a bit dumb for using `xmltodict` in the first place. 😅 Anyways, ping @AbletonDevTools/gotham-city

### CODE REVIEW

File: run_codenarc.py [hunk 1]
```diff
@@ -127,17 +128,17 @@ def _print_violations_in_packages(packages):
     # I believe that CodeNarc has a bug where it erroneously sets filesWithViolations
     # to the same value in every package. Therefore rather than looking at this attribute
     # value, we check to see if there are any File elements in the package.
-    for package in [p for p in packages if 'File' in p]:
+    for package in packages:
```
Comment (mst-ableton) [inline run_codenarc.py]: Why the change here? Additionally, we should change the comment above if it's no longer relevant.

Comment (nre-ableton) [reply run_codenarc.py]: Fixed in 46f7639faf17d1c85b3d064483ef9f302a9ae1b5.

File: run_codenarc.py [hunk 2]
```diff
@@ -124,20 +125,17 @@ def _print_violations_in_packages(packages):
     """
     num_violations = 0
 
-    # I believe that CodeNarc has a bug where it erroneously sets filesWithViolations
-    # to the same value in every package. Therefore rather than looking at this attribute
-    # value, we check to see if there are any File elements in the package.
-    for package in [p for p in packages if 'File' in p]:
+    for package in packages:
```
Comment (ala-ableton) [inline run_codenarc.py]: I don't understand how replacing `xmltodict` with `ElementTree` changes anything wrt. to the CodeNarc bug detailed above.

Comment (nre-ableton) [reply run_codenarc.py]: The comment refers to the fact that we couldn’t rely on that attribute to see if the package had any violations, and therefore we just had to loop and inspect each one. But because of the way that `xmltodict` worked we had to use the list comprehension to make sure this was safe and would not raise.

Admittedly the comment is confusing, and also not relevant anymore.
